core/templates/dev/head/domain/subtopic_viewer/SubtopicDataObjectFactory.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of frontend
+ * subtopic data domain objects.
+ */
+
+import { SubtitledHtmlObjectFactory } from
+  'domain/exploration/SubtitledHtmlObjectFactory';
+import { SubtopicPageContentsObjectFactory } from
+  'domain/topic/SubtopicPageContentsObjectFactory';
+import { RecordedVoiceoversObjectFactory } from
+  'domain/exploration/RecordedVoiceoversObjectFactory';
+import { VoiceoverObjectFactory } from
+  'domain/exploration/VoiceoverObjectFactory';
+
+angular.module('oppia').factory('SubtopicDataObjectFactory',
```

Comment:
@bansalnitish & @Showtim3, Should we ask contributors to do this in Angular way or it's fine to do it in AngularJS?

Reply:
Well the Angular way is surely better and required, since around 100-150 services are still left to be migrated and we are working on that. Also, since this is already done, I'd say we should only use the Angular way atleast for services, from now on.

Reply:
@Showtim3, Thanks for the suggestion! 

@nishantwrp, Can you please change this into Angular? (Sorry for the trouble!)

Reply:
@Showtim3, Thanks for the suggestion! 

@nishantwrp, Can you please change this into Angular? (Sorry for the trouble!)

Reply:
Done!